### PR TITLE
MINOR: Remove redundant clause in secureAclsEnabled check

### PR DESCRIPTION
**Pull request by ijuma**
Also include a few minor clean-ups.

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @guozhangwang

**Comment by ijuma** [comment]
Review by @guozhangwang

**Comment by omkreddy** [comment]
LGTM

**Comment by granthenke** [comment]
IMO this could benefit from having an associated jira so users could find the issue, change and its impact if there run into this. Otherwise, LGTM.

**Comment by ijuma** [comment]
I'll create the JIRA and add a note to `upgrade.html` as well.

**Comment by ijuma** [comment]
@granthenke, done.

**Comment by granthenke** [comment]
Thanks for the update @ijuma. LGTM

**Comment by ijuma** [comment]
Review by @gwenshap

File: core/src/main/scala/kafka/server/KafkaServer.scala [hunk 1]
```diff
@@ -269,28 +269,28 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
...
-      else
-        ""
+    val chrootIndex = config.zkConnect.indexOf("/")
+    val chrootOption = {
+      if (chrootIndex > 0) Some(config.zkConnect.substring(chrootIndex))
+      else None
     }
 
-    val secureAclsEnabled = JaasUtils.isZkSecurityEnabled() && config.zkEnableSecureAcls
+    val secureAclsEnabled = config.zkEnableSecureAcls
```
Comment (ewencp) [inline core/src/main/scala/kafka/server/KafkaServer.scala:280]: I think I'm confused about the original bug report. If `zookeeper.set.acl` is `true` but ZK security isn't enabled, then `secureAclsEnabled` will be `false && true` -> `false` and the if statement condition will be `true && !false` -> `true` and the exception will be thrown. Am I confused about something in the existing code? I don't see why the exception wouldn't be thrown.

**Comment by ewencp** [comment]
@ijuma I have a question about the original report (which would affect whether we really need the docs change), but otherwise this LGTM.

File: core/src/main/scala/kafka/server/KafkaServer.scala (hunk 1 above)
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaServer.scala:280]: You have good reason to be confused. The existing code has a redundant clause, but it's actually correct as well. My bad.

**Comment by ijuma** [comment]
@ewencp good catch, I closed the JIRA as Invalid, updated the PR title and removed the upgrade note.

**Comment by ewencp** [comment]
LGTM

**Pull request closed without merging**